### PR TITLE
New and better methods for Natural Alternating and Symmetric Groups

### PR DESCRIPTION
**Pull request by stevelinton**
as already existed  for Natural Symmetric Groups.

There might be more similar ommissions so don't merge yet.

### CODE REVIEW

**Comment by frankluebeck** [comment]
Looks good to me so far.

**Comment by stevelinton** [comment]
Looking at the code for natural symmetric and alternating groups more closely, it was a bit of a mess. There was a StabilizerOp method that could never be called, for instance. The latest version of this includes effective StabilizerOp methods for natural Sn for some basic actions. Still WIP.

**Comment by stevelinton** [comment]
As far as I know this is now good to merge.

**Comment by ChrisJefferson** [comment]
Only found one small issue:

```
gap> g := Group([()]);;
gap> IsAlternatingGroup(g);
true
gap> Size(g);
1/2
```

**Comment by alex-konovalov** [comment]
Thanks - let's merge this. I'd only update the title of this PR to reflect that is also contains some further improvements,not only Size and Random.

File: lib/gpprmsya.gi [hunk 1]
```diff
@@ -808,45 +817,214 @@ function ( G )
...
+    
+
+
+
+
+#############################################################################
+##
+#M  StabilizerOp( <nat-sym-grp>, ...... )
+##
+SYMGP_STABILIZER := function(sym, arg)
```
Comment (alex-konovalov) [inline lib/gpprmsya.gi]: @stevelinton while loading the library, `SYMGP_STABILIZER` causes "unbound global variable" warning for `gens` and `size`. This does not crash testinstall and teststandard, but hits the test of Tutorial examples. Shall I just add them to `local` ?

Comment (stevelinton) [inline lib/gpprmsya.gi]: Yes, please.

File: tst/teststandard/grpperm.tst [hunk 2]
```diff
@@ -92,6 +92,79 @@ gap> Collected(List(m,x->Index(w,x)));
...
+  (42,100), (43,100), (44,100), (45,100), (46,100), (47,100), (48,100), 
+  (49,100), (50,100), (51,100), (52,100), (53,100), (54,100), (55,100), 
+  (56,100), (57,100), (58,100), (59,100), (60,100), (61,100), (62,100), 
+  (63,100), (64,100), (65,100), (66,100), (67,100), (68,100), (69,100), 
+  (70,100), (71,100), (72,100), (73,100), (74,100), (75,100), (76,100), 
+  (77,100), (78,100), (79,100), (80,100), (81,100), (82,100), (83,100), 
+  (84,100), (85,100), (86,100), (87,100), (88,100), (89,100), (90,100), 
+  (91,100), (92,100), (93,100), (94,100), (95,100), (96,100), (97,100), 
+  (98,100), (99,100) ]
+gap> Centralizer(a, (1,2,3,4)(5,6,7,8)(9,10,11)(12,13,14));
```
Comment (alex-konovalov) [inline tst/teststandard/grpperm.tst]: @stevelinton this test does not run since `a` here. What shall I use for `a`?

**Comment by alex-konovalov** [comment]
Finally, something more non-trivial: this input, based on the `doc/tut/group.xml` leads to an error:
```
s8 := Group( (1,2), (1,2,3,4,5,6,7,8) );
a8 := DerivedSubgroup( s8 );
IsNaturalAlternatingGroup(a8);
pairs := Tuples( [1..8], 2 );;
IsTransitive( a8, pairs, OnPairs );
orbs := Orbits( a8, pairs, OnPairs );; Length( orbs );
u56 := Stabilizer( a8, orbs[2][1], OnPairs );; Index( a8, u56 );
```
and the error is
```
gap> u56 := Stabilizer( a8, orbs[2][1], OnPairs );; Index( a8, u56 );
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `AlternatingSubgroup' on 1 arguments called from
AlternatingSubgroup( CallFuncList( SYMGP_STABILIZER, Concatenation( [ SymmetricParentGroup( g ) ], arg 
     ) ) ) at /Users/alexk/GITREPS/gap/lib/gpprmsya.gi:1007 called from
orbish( G, pnt, gens, acts, act ) at /Users/alexk/GITREPS/gap/lib/oprt.gd:863 called from
CallFuncList( StabilizerFunc, arg ) at /Users/alexk/GITREPS/gap/lib/oprt.gi:2829 called from
<function "Stabilizer">( <arguments> )
 called from read-eval loop at line 30 of *stdin*
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk> 
```
If one removes the line `IsNaturalAlternatingGroup(a8);` then everything works.

File: tst/teststandard/grpperm.tst (hunk 2 above)
Comment (stevelinton) [inline tst/teststandard/grpperm.tst]: Oops! AlternatingGroup(14).

**Comment by stevelinton** [comment]
Fixed now in a commit on the master branch.

**Comment by alex-konovalov** [comment]
Thanks - https://github.com/gap-system/gap/commit/4485002a9c33729e6f70abc525b2e0c73f33e8ed fixed all.

**Pull request merged**